### PR TITLE
[Ray Client] Pickle task options for ray client

### PR DESCRIPTION
**Pull request by jjyao**
## Why are these changes needed?
We can just pickle task options instead of json so that we don't need to write custom `to_dict` and `from_dict` methods for complex python option objects (e.g. PlacementGroup).

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/util/client/common.py [hunk 1]
```diff
@@ -414,14 +414,7 @@ def set_task_options(task: ray_client_pb2.ClientTask,
...
 
-    # If there's a non-null "placement_group" in `options`, convert the
-    # placement group to a dict so that `options` can be passed to json.dumps.
-    pg = options.get("placement_group", None)
-    if pg and pg != "default":
-        options["placement_group"] = options["placement_group"].to_dict()
-
-    options_str = json.dumps(options)
-    getattr(task, field).json_options = options_str
+    getattr(task, field).pickled_options = pickle.dumps(options)
```
Comment (mwtian) [inline python/ray/util/client/common.py:417]: Curious, why not `task.field`?

**Review by mwtian — APPROVED** [review]

File: python/ray/util/client/common.py (hunk 1 above)
Comment (jjyao) [reply python/ray/util/client/common.py:417]: field is a string variable.

**Review by ckw017 — APPROVED** [review]
Looks good, can you bump this value as well https://github.com/ray-project/ray/blob/master/python/ray/util/client/__init__.py#L16

**Review by ericl — APPROVED** [review]

**Comment by jjyao** [comment]
Bumped the version and all tests passed. It's ready for merge.

**Pull request merged**